src/types/Client.d.ts

```diff
@@ -49,9 +49,24 @@ interface StreamApi
   document: StreamFn<SubscriptionEventHandlers>
 }
 
+interface MetricsResponse<T = object> {
+  value: T
+  metrics: {
+    'x-compute-ops': number
+    'x-byte-read-ops': number
+    'x-byte-write-ops': number
+    'x-query-time': number
+    'x-txn-retries': number
+  }
+}
+
 export default class Client {
   constructor(opts?: ClientConfig)
-  query<T = object>(expr: ExprArg, options?: QueryOptions): Promise<T>
+  query<T = object>(expr: ExprArg, options?: QueryOptions): Promise<T> | Promise<MetricsResponse<T>>
```

Comment:
This is a breaking change for TypeScirpt project :(

Reply:
Example:

![image](https://user-images.githubusercontent.com/2798204/160720548-653142a7-f623-46f3-a1d1-ed58bd4ba029.png)

```
error TS2322: Type 'boolean | MetricsResponse<boolean>' is not assignable to type 'boolean'.
```